### PR TITLE
[dendrite] Bump and correct postgresql subchart config

### PR DESCRIPTION
**Pull request by wrmilling**
**Description of the change**

* Bumped the PostgreSQL subchart version and image tag
* Corrected persistence definition in values.yaml

**Benefits**

Newest revision of PostgreSQL and correct values.yaml.

**Possible drawbacks**

N/A

**Applicable issues**

N/A

**Additional information**

N/A

**Checklist** 
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] 1Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [X] Variables have been documented in the `values.yaml` file.

### CODE REVIEW

**Comment by wrmilling** [comment]
@Jonnobrow Let me know if you have any qualms with the minor change here, tested via [wrmilling-charts](https://github.com/wrmilling/wrmilling-charts/tree/master/charts/dendrite) in [my cluster](https://github.com/wrmilling/k3s-gitops/blob/master/default/dendrite/dendrite.yaml)

**Comment by wrmilling** [comment]
Looks like this will have long term issues with #1572, only problem I saw is that you have to specify every component's database connection instead of being able to just supply username/password in a simplified configuration in the current form. Might be fodder for a future PR.

File: charts/incubator/dendrite/Chart.yaml [hunk 1]
```diff
@@ -82,6 +82,8 @@ dependencies:
 annotations:
   artifacthub.io/changes: |-
     - kind: changed
+      description: Corrected postgres persistence definition and bumped chart version
+    - kind: added
       description: Upgraded `common` chart dependency to version 4.5.0
```
Comment (bjw-s) [inline charts/incubator/dendrite/Chart.yaml:87]: Please remove the "old" changes, since those are not relevant for this specific version :)

**Review by bjw-s — APPROVED** [review]

**Review by bjw-s — CHANGES_REQUESTED** [review]
See earlier comment

File: charts/incubator/dendrite/Chart.yaml (hunk 1 above)
Comment (wrmilling) [reply charts/incubator/dendrite/Chart.yaml:87]: Sure thing, will get it updated as soon as I am back on my computer.

Comment (wrmilling) [reply charts/incubator/dendrite/Chart.yaml]: Updated.

**Review by bjw-s — APPROVED** [review]

**Pull request merged**